google-cloud-spanner/src/main/java/com/google/cloud/spanner/AbstractReadContext.java

```diff
@@ -380,7 +400,10 @@ void initTransaction() {
                   .setOptions(options)
                   .build();
           Transaction transaction =
-              rpc.beginTransaction(request, session.getOptions(), isRouteToLeader());
+              rpc.beginTransaction(
+                  request,
+                  getChannelHintOptions(session.getOptions(), getTransactionChannelHint()),
```

Comment:
We should not call `getChannelHintOptions(...)` for each request that is executed. Instead, this call should be moved to the constructor and called only once there. All of the fields involved are final, and the outcome of this calculation can also be final, so we should only do that once.

Reply:
Done, moved to constructor.